app/models/solid_cache/entry.rb

```diff
@@ -140,14 +140,25 @@ def to_binary(key)
 
         def expiry_candidate_ids(count, max_age:, max_entries:)
           cache_full = max_entries && max_entries < id_range
-          min_created_at = max_age.seconds.ago
 
-          uncached do
-            order(:id)
-              .limit(count * 3)
-              .pluck(:id, :created_at)
-              .filter_map { |id, created_at| id if cache_full || created_at < min_created_at }
-              .sample(count)
+          if max_age
+            min_created_at = max_age.seconds.ago
+
+            uncached do
+              order(:id)
+                .limit(count * 3)
+                .pluck(:id, :created_at)
+                .filter_map { |id, created_at| id if cache_full || created_at < min_created_at }
+                .sample(count)
+            end
+          elsif cache_full
```

Comment:
If we move the `cache_full` check before the `max_age` one, we can remove the `cache_full` check from the `filter_map`.

Something like:

```ruby
  if cache_full
    uncached do
      order(:id).limit(count * 3).pluck(:id).sample(count)
    end
  elsif max_age
    uncached do
      order(:id)
        .limit(count * 3)
        .pluck(:id, :created_at)
        .filter_map { |id, created_at| id if created_at < min_created_at }
        .sample(count)
    end
  else
    []
  end
```

Reply:
I went ahead and did this, but went further to try and DRY it up and show the intent of the two branches. Let me know if it doesn't pass muster.